udf/udf_enforce_date_param.sql

```diff
@@ -0,0 +1,39 @@
+/*
+
+Returns true if the two passed dates are identical or if either of them is
+equal to the special flag date 0001-01-01.
```

Comment:
The special flag date is 1111-11-11 here?

Reply:
Yes, it should be. I was initially using `0001-01-01` which is the earliest date supported by BQ, but decided 1111-11-11 is easier to type and still obscure enough to be a safe flag. Will fix.